Sources/CodegenCLI/Commands/Generate.swift

```diff
@@ -53,6 +61,36 @@ public struct Generate: ParsableCommand {
     }
   }
 
+  private func checkForCLIVersionMismatch() throws {
+    if case let .versionMismatch(cliVersion, apolloVersion) =
+        try VersionChecker.matchCLIVersionToApolloVersion(projectRootURL: rootOutputURL(for: inputs)) {
+      let errorMessage = """
+        Apollo Version Mismatch
+        We've detected that the version of the Apollo Codegen CLI does not match the version of the
+        Apollo library used in your project. This may lead to incompatible generated objects.
+
+        Please update your version of the Codegen CLI by following the instructions at:
+        https://www.apollographql.com/docs/ios/code-generation/codegen-cli/#installation
+
+        CLI version: \(cliVersion)
+        Apollo version: \(apolloVersion)
+        """
+
+      if ignoreVersionMismatch {
+        print("""
+          Warning: \(errorMessage)
+          """)
+      } else {
+
+        throw Error(errorDescription: """
+          Error: \(errorMessage)
+
+          To ignore this error and run the CLI anyways, use the argument: --ignore-version-mismatch.
```

Comment:
```suggestion
          To ignore this error during code generation, use the argument: --ignore-version-mismatch.
```
nit: Slightly clearer message of which command/workflow will ignore the error.

Reply:
Done